### PR TITLE
Vue3 + TS + Quasar - Implemented Auth functionality (Login & Signout)

### PR DESCRIPTION
**Pull request by AllanJeremy**
Added auth functionality.

## Overview
- Home page is now the auth page when a user isn't logged in
- Clicking the sign in with github link redirects the user to a redirecting page through the auth api url (`https://api.starter.dev/api/auth/signin`)
- The redirecting page sets the access token and redirects the user back to the homepage
- The home page contains a list of repositories & gists (dummy data for now)
- Clicking on the nav right hand corner dropdown then clicking signout clears the access token and redirects the user back to the login page

## Known bugs
None

### CODE REVIEW

**Review by ktrz — CHANGES_REQUESTED** [review]

File: vue3-ts-quasar/src/composables/auth/useAuth.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+interface SignoutRepsonse {
+  message: string;
+}
+
+//
+const { getRequest, postRequest } = apiHelper();
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/composables/auth/useAuth.ts:18]: Removed

File: vue3-ts-quasar/src/composables/auth/useToken.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const DEFAULT_TOKEN_STORAGE_NAME = 'vue3-starter-dev-access-token';
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/composables/auth/useToken.ts:3]: Removed

File: vue3-ts-quasar/src/composables/auth/useAuth.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    if (apiResponse?.access_token) {
+      saveAuthToken(apiResponse.access_token);
+    }
+
+    return apiResponse;
+  };
+
+  const signOut = () => {
+    removeAuthToken();
+    return postRequest<SignoutRepsonse>(
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/composables/auth/useAuth.ts:53]: Fixed

File: vue3-ts-quasar/src/composables/useNav.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  /** Redirect user the given link after the wait duration
+   * @param link The link to redirect to
+   * @param wait The amount of time to wait before redirecting in milliseconds
+   */
+  const goTo = (link: string, wait = 0) => {
+    if (window.location.pathname !== link) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      delay(() => {
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/composables/useNav.ts:20]: Done

File: vue3-ts-quasar/src/composables/useNav.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+  };
+
+  /** Redirect user an external link after the wait duration
+   * @param link The link to redirect to
+   * @param wait The amount of time to wait before redirecting in milliseconds
+   */
+  const goToExternal = (link: string, wait = 0) => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    delay(() => {
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/composables/useNav.ts:34]: Done

File: vue3-ts-quasar/src/helpers/api.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import axios, {
+  AxiosResponse,
+  AxiosRequestHeaders,
+  AxiosRequestConfig,
+} from 'axios';
+import { EnvironmentConfig } from '../config';
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/helpers/api.ts:8]: Done

File: vue3-ts-quasar/src/views/Redirect.vue [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+const { getToken } = useAuth();
+const { goToExternal } = useNav();
+
+const redirectIfTokenFound = async () => {
+  const { access_token } = await getToken();
+
+  if (!access_token) {
+    return setTimeout(redirectIfTokenFound, 1000);
+  }
```
[the remark this answers is not in the document]
Comment (AllanJeremy) [reply vue3-ts-quasar/src/views/Redirect.vue:30]: Done

File: vue3-ts-quasar/package.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "serve": "vue-cli-service serve",
+    "serve": "vue-cli-service serve --mode=development",
```
Comment (ktrz) [inline vue3-ts-quasar/package.json:6]: isn't serve in `development` mode by default?
https://cli.vuejs.org/guide/mode-and-env.html#modes

**Review by ktrz — APPROVED** [review]
LGTM
Left one small comment/question

**Pull request merged**